cmd/clusterctl/client/alias.go

```diff
@@ -41,6 +41,10 @@ type Template repository.Template
 // UpgradePlan defines a list of possible upgrade targets for a management group.
 type UpgradePlan cluster.UpgradePlan
 
+// CertManagerUpgradePlan defines the upgrade plan if cert-manager needs to be
+// upgraded to a different version.
+type CertManagerUpgradePlan cluster.CertManagerUpgradePlan
```

Comment:
Question for better understanding the PR:
Is there a particular reason why we are introducing a new type/a new client method vs extending the `UpgradePlan` type and the scope of `PlanUpgrade` func

Reply:
Yes. I was a bit unsure if adding a new type was the correct thing to do. 
I was initially trying to include it within the `UpgradePlan` type but it felt like I would be forcing the CertManager information within that type. Since the cert-manager is "cluster-level/global" entity  and the providers within the `UpgradePlan` correspond to a `CoreProvider` and `Contract`.
Also if I had included the information as part of the `UpgradePlan`, then printing it out would also be _interesting_ since it doesn't fit within the table.

This approach, although excessive (_in exposing another type and method on the clusterctl `Client`_), seemed a little more cleaner and decoupled from the other `PlanUpgrade` method.

Reply:
Thanks for the explanation, makes sense to me